syncd/syncd_flex_counter.cpp

```diff
@@ -522,9 +543,36 @@ void FlexCounter::flexCounterThread(void)
     while (m_runFlexCounterThread)
     {
         {
-            std::lock_guard<std::mutex> lock(g_mutex);
-            collectCounters(countersTable);
-            runPlugins(db);
+            std::map<std::string, std::vector<swss::FieldValueTuple>> portStats;
+            std::map<std::string, std::vector<swss::FieldValueTuple>> queueStats;
+            std::map<std::string, std::vector<swss::FieldValueTuple>> queueAttrs;
+            collectCounters(portStats, queueStats, queueAttrs);
+            if (m_pollInterval == PFC_WD_POLL_MSECS)
+            {
+                std::lock_guard<std::mutex> lock(g_mutex);
+                updateCounters(countersTable, portStats, queueStats, queueAttrs);
+                runPlugins(db);
+            }
+            else
+            {
+                for (auto& portStat : portStats)
+                {
+                    std::lock_guard<std::mutex> lock(g_mutex);
```

Comment:
Why is this guarded?
countersTable belongs only to this thread

Reply:
counterTable is the redis db shared by all threads and that's why we need the lock here.

Reply:
countersTable are the redis table shared by different thread, and that's why we need the lock

Reply:
From my point of view, different counterTable.set will write to the same redis db for same port/queue. Therefore I think we need to have lock for this.

Reply:
Redis handles that. It is single threaded and takes one request at a time.